### ISSUE TITLE
Lava Visiblity not working

### ISSUE THREAD

**Issue by XeR0x4** [maruohon/tweakeroo#267]
Even with toggling 'Lava Visibility' through the mod menu and with a hotkey, there is no visible change. Tried both respiration helmets & water breathing to no avail.  Does not work in Single- nor Multiplayer. Possible guess is it might have something to do with Optifine?

**Comment by maruohon** [comment]
This has been broken at least since 1.15. Currently if you enable the disableRenderDistanceFog option, you also get lava visibility for free.

**Comment by XeR0x4** [comment]
Cheers, could not find anything on the topic prior, apologies!

**Comment by aeniii** [comment]
I have the same issue but not even the disableRenderDistanceFog enabled works. I can't see anything more than three blocks. 
The rest tweakeroo stuff works fine.
Is there something i could try?

**Comment by maruohon** [comment]
@aeniii Do you have Optifine? AFAIK Optifine completely breaks all the fog options in Tweakeroo. Also I fixed this issue somewhat recently, but the fix is currently only in the version on my server: https://masa.dy.fi/mcmods/client_mods/?mcver=1.17.1 But as always it has not been tested with Optifine, and I would assume it's still broken with it.

**Comment by aeniii** [comment]
Yes i do play with optifine. I can't really play without...

**Comment by maruohon** [comment]
Why is that? If you mean for performance, then you would get way better rendering performance with Sodium. Or if you want to use shaders, then with Iris, which also is based on top of Sodium.

**Comment by aeniii** [comment]
Yeah i need it for performance. I tried the iris mod but without shaders, my setup can't handle that, it lagged even more with that and i couldn't find a good setting that's better than with optifine.

**Comment by aeniii** [comment]
Yeah i tried again with sodium and i get these huge FPS drops what i don't have with optifine. Is there anything i can do to get the lava visibility with optifine? Or is there a specific setting in sodium that i am missing about these frame drops?

**Comment by maruohon** [comment]
@aeniii You would probably get better help with Sodium from their Discord, I don't really know enough of the details and how different hardware is affected. All I can suggest is the more generic things like:
* What is your render distance set to in Sodium vs. while using Optifine? Do you get the same number of rendered chunks with both? (Optifine usually tends to render less from what I hear...) This would be the `C` value on the F3 screen on the 4th line.
* Check how your memory usage is doing (on the top right of the F3 screen), if you are constantly near the maximum allocation then you may get long garbage collection pauses. How much do you have allocated?
* Maybe try some of the options on the Advanced tab in the Sodium settings

As for the Tweakeroo options working with Optifine, that's pretty much just a `no`. I won't spend my time trying to debug and work around these incompatibility issues with a closed source mod, and it's likely it wouldn't even be possible.
